util/webhook/webhook.go

```diff
@@ -20,30 +22,48 @@ import (
 )
 
 type ArgoCDWebhookHandler struct {
-	ns               string
-	appClientset     appclientset.Interface
-	github           *github.Webhook
-	githubHandler    http.Handler
-	gitlab           *gitlab.Webhook
-	gitlabHandler    http.Handler
-	bitbucket        *bitbucket.Webhook
-	bitbucketHandler http.Handler
+	ns              string
+	appClientset    appclientset.Interface
+	github          *github.Webhook
+	gitlab          *gitlab.Webhook
+	bitbucket       *bitbucket.Webhook
+	bitbucketserver *bitbucketserver.Webhook
+	gogs            *gogs.Webhook
 }
 
 func NewHandler(namespace string, appClientset appclientset.Interface, set *settings.ArgoCDSettings) *ArgoCDWebhookHandler {
+
+	githubWebhook, err := github.New(github.Options.Secret(set.WebhookGitHubSecret))
+	if err != nil {
+		log.Warnf("Unable to init the Github webhook")
+	}
+	gitlabWebhook, err := gitlab.New(gitlab.Options.Secret(set.WebhookGitLabSecret))
+	if err != nil {
+		log.Warnf("Unable to init the Gitlab webhook")
```

Comment:
I think you should probably print the error here. E.g.
```golang
log.WithFields(log.Fields{"err": err}).Warn("Unable to init the Gitlab webhook")
```

Reply:
Sure, I will apply this change.